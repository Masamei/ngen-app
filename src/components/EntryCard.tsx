import { Entry } from '../interfaces'

export default function EntryCard({ entry }: { entry: Entry }) 
{
  function handleClick() {
    window.open(entry.Link, '_blank', 'noopener noreferrer')
  } 

  return (
    <div className='flex flex-wrap justify-evenly'>
      <div onClick={handleClick} 
          className='cursor-pointer text-silver bg-customBackground hover:bg-silver hover:text-green 
          rounded-lg p-4 m-2 min-w-52 max-w-52 w-52 transition duration-300 ease-in-out'>
        <h2 className='text-lg font-bold mt-2 elipsis-pretty'>{entry.API}</h2>
        <p className='text-gray-400 mb-2 text-[8px]'>Category: {entry.Category}</p>
        <p className='text-gray-400 pb-4 text-xs'>{entry.Description}</p>

        <span className='grid grid-cols-5 gap-1'>
          <p className={`text-gray-400 pb-4 text-[8px] ${entry.HTTPS ? 'text-green' : 'text-red line-through'}`}>HTTPS</p>
          <p className={`text-gray-400 pb-4 text-[8px] col-span-2`}>Cors: {entry.Cors}</p>
          {entry.Auth && <p className={`text-gray-400 pb-4 text-[8px] col-span-2`}>AUTH: {entry.Auth}</p>}
        </span>
      </div>
    </div>
  )
}
