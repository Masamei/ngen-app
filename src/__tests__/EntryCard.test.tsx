import { render } from '@testing-library/react'
import EntryCard from '../components/EntryCard'

const mockEntry = {
  API: 'The Muse',
  Description: 'Job board and company profiles',
  Auth: 'apiKey',
  HTTPS: true,
  Cors: 'unknown',
  Link: 'https://www.themuse.com/developers/api/v2',
  Category: 'Jobs',
};

describe('EntryCard Component', () => {
  test('renders EntryCard with correct data', () => {
    const { getByText } = render(<EntryCard entry={mockEntry} />)
    
    expect(getByText('The Muse')).toBeInTheDocument()
    expect(getByText('Job board and company profiles')).toBeInTheDocument()
    expect(getByText('Cors: unknown')).toBeInTheDocument()
  })

  test('matches snapshot', () => {
    const { asFragment } = render(<EntryCard entry={mockEntry} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
