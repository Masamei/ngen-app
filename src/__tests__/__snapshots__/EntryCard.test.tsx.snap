// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryCard Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class='max-w-sm rounded overflow-hidden shadow-lg p-4 bg-white'
  >
    <div
      class='font-bold text-xl mb-2'
    >
      The Muse
    </div>
    <p
      class='text-gray-700 text-base mb-2'
    >
      Job board and company profiles
    </p>
    <p
      class='text-gray-600 text-sm mb-2'
    >
      <strong>
        Auth:
      </strong>
       apiKey
    </p>
    <p
      class='text-gray-600 text-sm mb-2'
    >
      <strong>
        HTTPS:
      </strong>
       Yes
    </p>
    <p
      class='text-gray-600 text-sm mb-2'
    >
      <strong>
        Cors:
      </strong>
       unknown
    </p>
    <p
      class='text-gray-600 text-sm mb-2'
    >
      <strong>
        Category:
      </strong>
       Jobs
    </p>
    <a
      class='text-blue-500 hover:text-blue-700'
      href='https://www.themuse.com/developers/api/v2'
      rel='noopener noreferrer'
      target='_blank'
    >
      Visit API
    </a>
  </div>
</DocumentFragment>
`;
